Userland/Libraries/LibWebContent/RequestServerAdapter.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, Dex♪ <dexes.ttp@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/File.h>
+#include <LibProtocol/Request.h>
+#include <LibProtocol/RequestClient.h>
+#include <LibWebContent/RequestServerAdapter.h>
+
+namespace WebContent {
+
+ErrorOr<NonnullRefPtr<RequestServerRequestAdapter>> RequestServerRequestAdapter::try_create(NonnullRefPtr<Protocol::Request> request)
+{
+    return adopt_nonnull_ref_or_enomem(new (nothrow) RequestServerRequestAdapter(move(request)));
+}
+
+RequestServerRequestAdapter::RequestServerRequestAdapter(NonnullRefPtr<Protocol::Request> request)
+    : m_request(request)
+{
+    request->on_buffered_request_finish = [weak_this = make_weak_ptr()](auto success, auto total_size, auto const& response_headers, auto response_code, auto payload) {
+        weak_this->on_buffered_request_finish(success, total_size, response_headers, response_code, move(payload));
+    };
+
+    request->on_certificate_requested = [weak_this = make_weak_ptr()]() {
+        auto certificate_and_key = weak_this->on_certificate_requested();
```

Comment:
if weak_this is null here (i.e., the request was already destroyed by the time the callback is called), won't this crash? This should do one of those `if (auto strong_this = weak_this.strong_ref())` dances, right?